changelog/upgrade-notes.md

```diff
@@ -33,6 +33,10 @@ Users compiling their code with the library should note that the following typed
 * ol.style.RegularShapeRenderOptions to ol.RegularShapeRenderOptions
 * ol.style.StyleFunction to ol.StyleFunction
 
+#### KML format now uses URL()
+
+The URL constructor is supported by all modern browsers, but not by older ones, such as IE. To use the KML format in such older browsers, a URL polyfill will have to be loaded before use.
```

Comment:
> But isn't it also translating a relative url into an absolute url in the case of the KML format?

the only logic for which `goog.Uri` is used (in 4 places in KML format, mainly for styles/stylemaps, but also for other links) is:

if string is absolute url (starts with http or https), use that, else append string to node.baseURI

This could be added, to the KML format or perhaps better to `ol.uri`. Unlike the tests, which parse a url into its constituent parts, the KML format doesn't need any real parsing of the url other than determining if it's absolute or relative.

I don't think using an npm package is a good idea, as it penalises the majority using modern browsers (extra code overhead) just to support the minority using old browsers.